java/ql/src/AlertSuppressionAnnotations.ql

```diff
@@ -0,0 +1,107 @@
+/**
+ * @name Alert suppression using annotations
+ * @description Generates information about alert suppressions
+ *              using 'SuppressWarnings' annotations.
+ * @kind alert-suppression
+ * @id java/alert-suppression-annotations
+ */
+
+import java
+import Metrics.Internal.Extents
+
+/** Gets the LGTM suppression annotation text in the string `s`, if any. */
+bindingset[s]
+string getAnnotationText(string s) {
+  // match `lgtm[...]` anywhere in the comment
+  result = s.regexpFind("(?i)\\blgtm\\s*\\[[^\\]]*\\]", _, _)
+}
+
+/**
+ * An alert suppression annotation.
+ */
+class SuppressionAnnotation extends SuppressWarningsAnnotation {
+  string text;
+
+  SuppressionAnnotation() {
+    text = this.getASuppressedWarningLiteral().getValue() and
+    exists(getAnnotationText(text))
+  }
+
+  /**
+   * Gets the text of this suppression annotation.
+   */
+  string getText() { result = text }
+
+  private Annotation getASiblingAnnotation() {
+    result = getAnnotatedElement().(Annotatable).getAnAnnotation() and
+    (getAnnotatedElement() instanceof Callable or getAnnotatedElement() instanceof RefType)
+  }
+
+  private Annotation firstAnnotation() {
+    exists(Annotation m, int i |
+      result = m and
+      m = getASiblingAnnotation() and
+      i = rankOfAnnotation(m) and
+      not exists(Annotation other | other = getASiblingAnnotation() | rankOfAnnotation(other) < i)
```

Comment:
This can be simplified to
```
result = min(this.getASiblingAnnotation() as m
  order by m.getLocation().getStartLine(), m.getLocation().getStartColumn())
```
and that should also perform better.